go/arrow/array/builder.go

```diff
@@ -279,15 +279,6 @@ func NewBuilder(mem memory.Allocator, dtype arrow.DataType) Builder {
 	case arrow.TIME64:
 		typ := dtype.(*arrow.Time64Type)
 		return NewTime64Builder(mem, typ)
-	case arrow.INTERVAL:
-		switch dtype.(type) {
-		case *arrow.DayTimeIntervalType:
-			return NewDayTimeIntervalBuilder(mem)
-		case *arrow.MonthIntervalType:
-			return NewMonthIntervalBuilder(mem)
-		case *arrow.MonthDayNanoIntervalType:
-			return NewMonthDayNanoIntervalBuilder(mem)
-		}
```

Comment:
looks like you removed various code related to INTERVAL, was this intentional?

Reply:
Yea, if you look at `datatype.go` you'll see that the `INTERVAL` enum value was deprecated and said it would be removed in the next Major release in favor of the individual interval IDs.

Reply:
i don't think it needs to be a separate PR for this